### ISSUE TITLE
DataStore - ObserveQuery deleted items are showing

### ISSUE THREAD

**Issue by a-alen** [aws-amplify/amplify-android#2069]
### Before opening, please confirm:

- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-android/issues?q=is%3Aissue+) and [discussions](https://github.com/aws-amplify/amplify-android/discussions).

### Language and Async Model

Kotlin

### Amplify Categories

GraphQL API, DataStore

### Gradle script dependencies

<details>

```groovy
// Put output below this line
    def amplify_version = "1.37.5"

    implementation "com.amplifyframework:core:$amplify_version"
    implementation "com.amplifyframework:aws-api:$amplify_version"
    implementation "com.amplifyframework:aws-auth-cognito:$amplify_version"
    implementation "com.amplifyframework:aws-datastore:$amplify_version"
    coreLibraryDesugaring 'com.android.tools:desugar_jdk_libs:1.1.6'

```

</details>

### Environment information

<details>

```
# Put output below this line
distributionUrl=https\://services.gradle.org/distributions/gradle-7.5.1-bin.zip

```

</details>

### Describe the bug

When item is deleted with Amplify.DataStore.delete, that deleted item stays in Amplify.DataStore.observeQuery. You need to exit from app, and when you start it again ObserveQuery shows correct number of items, until next delete.

Column _deleted is not accessible, so it is not possible to filter that item.

### Code Snippet

```java
// Put your code below this line.

```

### Log output

<details>

```
// Put your logs below this line
2022-10-27 20:54:22.699  5341-5409  AWS_DATASTORE_GIT       com.example.barcodelogin             I  OBSERVE STARTED
2022-10-27 20:54:22.727  5341-5417  AWS_DATASTORE_GIT       com.example.barcodelogin             I  OBSERVE number of items: 8
2022-10-27 20:54:22.755  5341-5441  AWS_DATASTORE_GIT       com.example.barcodelogin             I  QUERY number of items: 8
2022-10-27 20:54:39.407  5341-5493  AWS_DATASTORE_GIT       com.example.barcodelogin             I  OBSERVE number of items: 8
2022-10-27 20:54:39.407  5341-5493  AWS_DATASTORE_GIT       com.example.barcodelogin             I  ITEM DELETED
2022-10-27 20:54:39.416  5341-5495  AWS_DATASTORE_GIT       com.example.barcodelogin             I  QUERY number of items: 7
2022-10-27 20:54:41.409  5341-5494  AWS_DATASTORE_GIT       com.example.barcodelogin             I  OBSERVE number of items: 8
2022-10-27 20:54:41.417  5341-5515  AWS_DATASTORE_GIT       com.example.barcodelogin             I  QUERY number of items: 7

```

</details>

### GraphQL Schema

<details>

```graphql
// Put your schema below this line


```

</details>

**Comment by AnilMaktala** [comment]
Hey @a-alen 👋, Thanks for raising the issue. We can replicate the issue and marked this as a bug for the team to evaluate further.

**Comment by a-alen** [comment]
Hi, also - when there is a update mutation inside the application, ObserveQuery does not transmit/reacts to that data change. Is this wanted behavior or is it a bug? If it is a bug, should I open it as a new issue?

**Comment by admin-ssg** [comment]
Hello, is there any progresses on this? :)

**Comment by iooi** [comment]
+1!! Any update??

**Comment by iooi** [comment]
also reproduced on iOS.

**Comment by Makatun** [comment]
+1

**Comment by CyprienRicque** [comment]
Same problem here. Do you have any good workaround?

**Comment by Gamer2015** [comment]
+1

**Comment by ankpshah** [comment]
Hello, we merged #2522 which should resolve this issue when released.  Will communicate the updated version number so that you can test and make sure the issue is fully fixed.

**Comment by ankpshah** [comment]
This has been fixed in version `2.11.1`
